addons/imaginary.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
+
+# T&M Hansson IT AB © - 2023, https://www.hanssonit.se/
+# GNU General Public License v3.0
+# https://github.com/nextcloud/vm/blob/master/LICENSE
+
+true
+SCRIPT_NAME="Imaginary Docker"
+SCRIPT_EXPLAINER="This script will install Imaginary which is a replacement for the less secure Imagick.
+It can speedup the loading of previews in Nextcloud a lot."
+# shellcheck source=lib.sh
+source /var/scripts/fetch_lib.sh
+
+# Check for errors + debug code and abort if something isn't right
+# 1 = ON
+# 0 = OFF
+DEBUG=0
+debug_mode
+
+# Check if root
+root_check
+
+# Compatible with NC24 and above
+lowest_compatible_nc 24
+
+# Only applies if previewgenerator is installed
+if ! is_app_installed previewgenerator
+then
+    msg_box "Imaginary is only needed if your use the Preview Generator app, no need to run this script."
+    exit
+fi
+
+# Check if Imaginary is already installed
+if ! does_this_docker_exist nextcloud/aio-imaginary
+then
+    # Ask for installing
+    install_popup "$SCRIPT_NAME"
+else
+    # Ask for removal or reinstallation
+    reinstall_remove_menu "$SCRIPT_NAME"
+    # Removal
+    if yesno_box_no "Do you want to remove the Imaginary docker container and settings?"
+    then
+        docker-compose_down "$SCRIPTS"/imaginary-docker/docker-compose.yml
+        nextcloud_occ config:system:delete enabledPreviewProviders
+        nextcloud_occ config:system:delete preview_imaginary_url
+        rm -rf "$SCRIPTS"/imaginary-docker
+    fi
+    # Show successful uninstall if applicable
+    removal_popup "$SCRIPT_NAME"
+fi
+
+# Generate docker-compose.yml
+mkdir -p "$SCRIPTS"/imaginary-docker
+if [ ! -f "$SCRIPTS/imaginary-docker/docker-compose.yml" ]
+then
+    touch "$SCRIPTS/imaginary-docker/docker-compose.yml"
```

Comment:
Why use docker-compose for this? I wouldnt say it is necessary here

Reply:
I think it's a cleaner solution.

Reply:
But yeah, maybe not, since we actually have a `funtion` for stopping... Will think about this.